File: core/AddDigits/AddDigits.java
package com.dbc;

public class AddDigits {
    public int addDigits(int num) {
        while (num >= 10) {
            int temp = 0;
            while (num != 0) {
                temp += num % 10;
                num = num / 10;
            }
            num = temp;
        }
        return num;
    }
}

File: core/AddDigits/AddDigits.py
def addDigits(num: int) -> int:
    while num >= 10:
        temp = 0
        while num != 0:
            temp += num % 10
            num = num // 10

        num = temp

    return num

if __name__ == '__main__':
    print(addDigits(10))
